### PR TITLE
Reduce pointer usage

### PR DESCRIPTION
**Pull request by ChrisThrasher**
## Description

Continuing what I started in https://github.com/SFML/SFML/pull/3046

### CODE REVIEW

**Review by vittorioromeo — APPROVED** [review]
LGTM, good simplification.

**Comment by ChrisThrasher** [comment]
SFMLBuildMaster: Build this please.

File: examples/android/app/src/main/jni/main.cpp [hunk 1]
```diff
@@ -26,53 +26,53 @@
 int vibrate(sf::Time duration)
 {
     // First we'll need the native activity handle
-    ANativeActivity* activity = sf::getNativeActivity();
+    ANativeActivity& activity = *sf::getNativeActivity();
 
     // Retrieve the JVM and JNI environment
-    JavaVM* vm  = activity->vm;
-    JNIEnv* env = activity->env;
+    JavaVM& vm  = activity.vm;
```
Comment (kimci86) [inline examples/android/app/src/main/jni/main.cpp:32]: I don't think this compiles. `vm` is still a pointer.
For reference: [ANativeActivity structure reference](https://developer.android.com/ndk/reference/struct/a-native-activity#struct_a_native_activity_1a71db04679afeb8c87ba18d398a7f9187)

**Review by kimci86 — CHANGES_REQUESTED** [review]

File: examples/android/app/src/main/jni/main.cpp (hunk 1 above)
Comment (ChrisThrasher) [reply examples/android/app/src/main/jni/main.cpp]: Oh I see. I assumed this file was being compiled in CI but I don't believe it is.

**Comment by ChrisThrasher** [comment]
SFMLBuildMaster: Build this please.

**Pull request merged**